src/ng-select/lib/selection-model.ts

```diff
@@ -62,14 +64,25 @@ export class DefaultSelectionModel implements SelectionModel {
     }
 
     private _setChildrenSelectedState(children: NgOption[], selected: boolean) {
-        children.forEach(x => x.selected = selected);
+        children.forEach(x => {
+            if (x.disabled !== true) {
+                x.selected = selected;
+            }
+        });
     }
 
     private _removeChildren(parent: NgOption) {
-        this._selected = this._selected.filter(x => x.parent !== parent);
+        this._selected = [
+            ...this._selected.filter(x => x.parent !== parent), 
+            ...parent.children.filter(x => x.disabled === true && x.selected === true)
+        ];
     }
 
     private _removeParent(parent: NgOption) {
         this._selected = this._selected.filter(x => x !== parent)
     }
+
+    private _inactiveChildren(item: NgOption): boolean {
+        return item.children.find(x => x.disabled === true && x.selected !== true) !== undefined;
```

Comment:
this can be change to:
```suggestion
        return item.children.some(x => x.disabled && !x.selected);
```
or even and then invert to `_activeChildren`
```suggestion
        return item.children.every(x => !x.disabled || x.selected);
```

Reply:
Done